uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/SearchRESTController.java

```diff
@@ -0,0 +1,224 @@
+import org.apereo.portal.portlets.lookup.PersonLookupHelperImpl;
+import org.apereo.portal.rest.utils.PortletRegistryUtil;
+import org.apereo.portal.security.IPerson;
+import org.apereo.portal.security.IPersonManager;
+import org.apereo.services.persondir.IPersonAttributes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+/**
+ * REST Search endpoint that aggregates search of various types.
+ *
+ * <p>JSON results example (whitespace added for clarity):
+ *
+ * <pre><code>
+ * {
+ *   "people": [
+ *     {
+ *       "telephoneNumber": [
+ *         "(555) 555-5555"
+ *       ],
+ *       "mail": [
+ *         "amy.administrator@university.edu"
+ *       ],
+ *       "displayName": [
+ *         "Amy Administrator"
+ *       ],
+ *       "givenName": [
+ *         "Amy"
+ *       ],
+ *       "sn": [
+ *         "Administrator"
+ *       ],
+ *       "title": [
+ *         "Portal Administrator"
+ *       ],
+ *       "department": [
+ *         "IT Services"
+ *       ],
+ *       "username": [
+ *         "admin"
+ *       ]
+ *     },
+ *     {
+ *       "telephoneNumber": [
+ *         "(555) 555-5555"
+ *       ],
+ *       "mail": [
+ *         "samuel.staff@example.org"
+ *       ],
+ *       "displayName": [
+ *         "Samuel Staff"
+ *       ],
+ *       "givenName": [
+ *         "Samuel"
+ *       ],
+ *       "sn": [
+ *         "Staff"
+ *       ],
+ *       "title": [
+ *         "Database Administrator"
+ *       ],
+ *       "department": [
+ *         "IT Services"
+ *       ],
+ *       "username": [
+ *         "staff"
+ *       ]
+ *     }
+ *   ],
+ *   "portlets": [
+ *     {
+ *       "description": "Campus News",
+ *       "fname": "campus-news",
+ *       "name": "Campus News",
+ *       "title": "Campus News",
+ *       "url": "/uPortal/f/news-fav-collection/p/campus-news.u33l1n8/max/render.uP"
+ *     },
+ *     {
+ *       "description": "Top stories from The Chronicle of Higher Education: Wired Campus Edition",
+ *       "fname": "chronicle-wired",
+ *       "name": "The Chronicle: Wired Campus",
+ *       "title": "The Chronicle: Wired Campus",
+ *       "url": "/uPortal/f/news-fav-collection/p/chronicle-wired.u33l1n9/max/render.uP"
+ *     },
+ *     {
+ *       "description": "Calendar of campus, academic, and personal events.",
+ *       "fname": "calendar",
+ *       "name": "Calendar",
+ *       "title": "Calendar",
+ *       "url": "/uPortal/f/welcome/p/calendar.u17l1n13/max/render.uP"
+ *     }
+ *   ]
+ * }
+ * </code></pre>
+ *
+ * @since 5.0
+ */
+@Controller
+@RequestMapping("/search")
```

Comment:
We already have a lot of REST endpoints, and we're likely to be adding several more in the next handful of releases. 
 We really must use URLs that convey more nuance, and ideally we adopt (and follow) a pattern for naming them.

Here is one of the more recent:  `@RequestMapping(value = "/v4-3/dlm/portletRegistry.json", method = RequestMethod.GET)`

Now then -- I'm not wedded to appending `.json` to the end of them (it seems like the last time we discussed it we were leaning against it).  I do kinda like having the version in there.

What about `/v5-0/portal/search`?  It includes the version, and signals that it's not about searching for groups or layout nodes... it's about the central uPortal search feature.

Reply:
Sounds good!